code_conventions.md

```diff
@@ -172,156 +125,46 @@ for (int i = 0; i < len; i++)
---------
-Example for a file CuraEngine/src/folder/SomeClass.h (UpperCamelCase):
-~~~~~~~~~~~~~~~{.cpp}
-#ifndef (FOLDER_SOME_CLASS_H)
-#define FOLDER_SOME_CLASS_H
-
-...
-
-#endif //FOLDER_SOME_CLASS_H
-~~~~~~~~~~~~~~~
-Each header file must include a header guard as shown above. The defined macro is adopted from the path and name of the class and must follow the rules for macros (UPPER_CASE).
-Here the folder `src` is skipped, because all header and implementation files of CuraEngine are in `src`.
-
-Pointers and References C(++)
-----
-When declaring a variable the asterisk (*) and the ampersand (&) should be connected to the type, and there should be a space between it and the variable name.
-~~~~~~~~~~~~~~~{.cpp}
-Object& instance; // allowed
-Object &instance; // not allowed
-~~~~~~~~~~~~~~~
-Multiple declarations are allowed, as long as the objects have the same type
-~~~~~~~~~~~~~~~{.cpp}
-Object instance, other; // allowed
-Object* instance, other; // not allowed, the other variable is of type Object, rather than Object*
-~~~~~~~~~~~~~~~
-
-
-Null pointer (C++)
-----
-For C++, never use `NULL`, but use `nullptr` instead. NULL is an integer, not a pointer.
+```
 
-Ordering
-----
-* Members: Implement functions Top-Down, starting with constructors/deconstructors in case of classes. That way, a class implementation can be read as a page from a book: from top to bottom providing clarity. This can be an issue for declarative language constructs like C/C++ , but then it's good practice to use forward declarations.
-* Go from public, protected to private. Reasoning behind this is similar. When using a class, one is more interested in the public items. For inheriting the protected ones can be interesting, while the private parts should only be meaningfull to the maintainer of the class. This helps with the OOP paradigm of implementation hiding.
+## Ordering
+Functions SHOULD be implemented Top-Down, starting with constructors/deconstructors (in case of classes). That way, a class implementation can be read as a page from a book: from top to bottom. This can be an issue for declarative language constructs like C/C++ , but it's good practice to use forward declarations in the first place.
 
-Strings
-----
-Strings are double quotes. While python and php allow single and double quoted strings, and PEP8 only says "pick a rule and stick to it". We decided to do double quotes to match C++.
+Go from public, protected to private. When using a class, one is usually more interested in the public items. For inheriting the protected ones can be interesting, while the private parts should only be meaningful to the maintainer of the class itself.
 
-Lines and Linelength
-----
-* "Maximum line length" - There is no hard limit on the line length, but as a thumb of rule, try to keep it to at most 120 characters. This helps doing the code reviews!
-* A single line contains only one statement
+## Strings
+Strings MUST be defined by using double quotes.
```

Comment:
I disagree that this is in the general conventions. There are languages where the differences between single and double quotes does have a meaning (usually, one does string substitutions and the other does not) and enforcing double quotes for those languages ends up with performance penalties or similar issues.

Reply:
Have you read the rest of the document? This is literally stated; Language specific ones can override general ones. So if there is a reason to do so, leave that up to that languages codestyle.